flexo/src/mirror_flexo.rs

```diff
@@ -590,7 +590,27 @@ fn cfs_path_from_pkg_path(path: &Path) -> Option<PathBuf> {
 pub struct DownloadOrder {
     /// This path is relative to the given root directory.
     pub requested_path: StrPath,
-    pub id: Uuid,
+    pub non_cacheable_unique_id: Option<Uuid>,
+}
+
+impl DownloadOrder {
+    pub fn new(requested_path: StrPath) -> Self {
+        if !(requested_path.to_str().ends_with(".db") ||
+            requested_path.to_str().ends_with(".db.sig") ||
+            requested_path.to_str().ends_with(".files") ||
+            requested_path.to_str().ends_with(".files.sig")) {
+            Self {
+                requested_path: requested_path,
```

Comment:
Rust supports the same shorthand syntax like JavaScript and TypeScript if the variable name matches the name of the property:

```suggestion
                requested_path,
```